onnxruntime/core/providers/cpu/controlflow/if.cc

```diff
@@ -139,6 +142,24 @@ Status IfImpl::Initialize() {
     subgraph_output_names_.push_back(output->Name());
   }
 
+  auto num_implicit_inputs = implicit_inputs_.size();
+  subgraph_feed_names_.reserve(num_implicit_inputs);
+  subgraph_feeds_mlvalue_idxs_.reserve(num_implicit_inputs);
+
+  auto& ort_value_name_idx_map = session_state_.GetOrtValueNameIdxMap();
+
+  // pass in implicit inputs as feeds.
+  for (auto& entry : implicit_inputs_) {
+    // prune to values that are in this subgraph as the implicit inputs cover both 'then' and 'else' subgraphs.
+    // alternatively we could track implicit inputs on a per-attribute basis in the node, but that
+    // would make that tracking a bit more complicated.
+    int idx;
+    if (ort_value_name_idx_map.GetIdx(entry.first, idx).IsOK()) {
+      subgraph_feed_names_.push_back(entry.first);
+      subgraph_feeds_mlvalue_idxs_.push_back(idx);
+    }
+  }
+
   auto status = AllocateOutputTensors();
```

Comment:
Previously this was done in CreateFeedsFetchesManager which was called once and then cached.

Now you're doing this every time the If operator's Compute method is called. 

Isn't that worse?